### ISSUE TITLE
Webcam Video effects camera not detected  in online meet

### ISSUE THREAD

**Issue by Arjun31415** [intermezzio/webcam-video-effects#4]
So I have followed all the instructions for setup and usage and the effects work  with desktop applications such as Teams and Discord.
However, it does not work in online meets such at Google Meet and Teams Online

**Comment by intermezzio** [comment]
When using the camera on browsers, it's important to start the shell script before opening the call in your browser or the browser won't detect the webcam. Specifically, make sure you start the scripts in this order:
1. Turn the webcam on
2. Start the shell script that sends data to the webcam
3. Open the call on your browser

Once the browser is opened and it identifies the webcam, you can stop and restart the script as much as you'd like, but it has to be streaming data when the call starts.

**Comment by Arjun31415** [comment]
Yes I followed the same steps, Firefox says unable to connect to webcam and chrome just keeps loading the webcam but there is no video

**Comment by Arjun31415** [comment]
> That's expected - Changing the video_nr should make the new webcam you play from `/dev/video5`. That's what the `-o 5` command does. Can you confirm that this is the case by pasting the output of `v4l2-ctl --list-devices`? You should get your existing webcams plus "Webcam Video Effects"

the output is 
```
Webcam Video Effects (platform:v4l2loopback-000):
	/dev/video5

HP Wide Vision HD Camera: HP Wi (usb-0000:06:00.3-3):
	/dev/video0
	/dev/video1
```
it does not seem like /dev/video2 is the default webcam device

**Comment by Arjun31415** [comment]
> > > That's expected - Changing the video_nr should make the new webcam you play from `/dev/video5`. That's what the `-o 5` command does. Can you confirm that this is the case by pasting the output of `v4l2-ctl --list-devices`? You should get your existing webcams plus "Webcam Video Effects"
> > 
> > 
> > the output is
> > ```
> > Webcam Video Effects (platform:v4l2loopback-000):
> > 	/dev/video5
> > 
> > HP Wide Vision HD Camera: HP Wi (usb-0000:06:00.3-3):
> > 	/dev/video0
> > 	/dev/video1
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > it does not seem like /dev/video2 is the default webcam device
> 
> This is actually a good sign! It seems your webcam works as intended. The config I asked you to change makes `/dev/video5` the default instead of `/dev/video2`. It seems that wasn't a cause of the problem though. You can switch to `video_nr=2` to get it back on `/dev/video2` if you want, but it really doesn't make much of a difference.
> 
> > Also when using google Meet it does not ask permission for `Webcam Video Effects` it asks instead for the default webcam ![image](https://user-images.githubusercontent.com/72289243/136882575-167ae22f-850b-4bf6-ae27-e1782a183546.png)
> 
> When using Google meet, by default it doesn't use Webcam Video effects so you'll have to change that in settings. First go to the meet as shown here:
> 
> ![image](https://user-images.githubusercontent.com/31703249/136884244-51b7d9b8-dd65-40cb-835e-60aa04d5bf7c.png)
> 
> Then click Settings > Video and select "Webcam Video Effects" from the dropdown (make sure you are running `video_looper_complete.sh` when you're at this page - otherwise start the script and refresh the google meet)
> 
> ![image](https://user-images.githubusercontent.com/31
703249/136884685-3d9dfabe-432d-47e2-b4ad-9bb8ed5f98cb.png)
> 
> Then choose the camera you want and it should ask your permission to use it.
> 
> ![image](https://user-images.githubusercontent.com/31703249/136884723-f6638647-db7c-4be8-bcae-10abeef7d9d9.png)
> 
> Let me know if these steps work for you!

When i try to click on the dropdown 
![image](https://user-images.githubusercontent.com/72289243/136886651-c4e2e878-00e3-4ddf-a79a-d7d90a588d43.png)

The other webcams are not listed

**Comment by intermezzio** [comment]
Can you run the video looping script
```sh
./video_looper_complete.sh -t "test" -b "webcam" # add "-o 5" if you kept /dev/video5 as your default
```
and then go to https://webcamtests.com/ and check if any of the webcams display Webcam Video Effects?

**Comment by Arjun31415** [comment]
and while reading through your code, I found that there is no `-vodec`flag could it be because of that

**Comment by intermezzio** [comment]
Thanks for the information! I'll try to figure out if it's related to that, the `-re` flag, or something else entirely. In the meantime, when you run the `video_looper_complete.sh` script, while it's running you can run `ps aux | grep ffmpeg`, copy the command that the script is outputting, and add any flags you want before rerunning it.

I'm still working on fixing the blur options - that's my newest feature, and I'm still trying to fix it (maybe adding a fade-in blur and changing the way blurs are calculated). If you find any other interesting information regarding the commands please let me know in this issue!

**Comment by Arjun31415** [comment]
> Thanks for the information! I'll try to figure out if it's related to that, the `-re` flag, or something else entirely. In the meantime, when you run the `video_looper_complete.sh` script, while it's running you can run `ps aux | grep ffmpeg`, copy the command that the script is outputting, and add any flags you want before rerunning it.
> 
> I'm still working on fixing the blur options - that's my newest feature, and I'm still trying to fix it (maybe adding a fade-in blur and changing the way blurs are calculated). If you find any other interesting information regarding the commands please let me know in this issue!

I ran `./video_looper_complete.sh -v "sample.mp4"` then `ps aux | grep ffmpeg` in another terminal which gave-

```
36055 33.3  0.7 1545136 110576 pts/2  SLl+ 09:42   0:23 ffmpeg -stream_loop -1 -re -i stream.mp4 -filter_complex  [0]   format=yuv420p[v] -map [v] -f v4l2 /dev/video2
36758  0.0  0.0  17636   844 pts/3    S+   09:43   0:00 grep --color=auto ffmpeg
```

**Comment by intermezzio** [comment]
The should be fixed in [v0.2.0](https://github.com/intermezzio/webcam-video-effects/releases/tag/v0.2.0) as blurred backgrounds were fully rewritten. If there are still problems, please open another issue.
